gnosis_vpn-lib/src/config/v2.rs

```diff
@@ -0,0 +1,369 @@
+use std::cmp::PartialEq;
+use std::collections::HashMap;
+use std::net::SocketAddr;
+use std::time::Duration;
+use std::vec::Vec;
+
+use serde::{Deserialize, Serialize};
+use url::Url;
+
+use crate::connection::{Destination as ConnDestination, SessionParameters};
+use crate::entry_node::EntryNode;
+use crate::peer_id::PeerId;
+use crate::session;
+use crate::wireguard::config::{Config as WireGuardConfig, ManualMode as WireGuardManualMode};
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+pub struct Config {
+    pub version: u8,
+    hoprd_node: HoprdNode,
+    destinations: Option<HashMap<PeerId, Destination>>,
+    connection: Option<Connection>,
+    wireguard: Option<WireGuard>,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+struct HoprdNode {
+    endpoint: Url,
+    api_token: String,
+    internal_connection_port: Option<u16>,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+struct Destination {
+    meta: Option<HashMap<String, String>>,
+    path: Option<DestinationPath>,
+}
+
+#[derive(Clone, Debug, PartialEq, Serialize, Deserialize)]
+enum DestinationPath {
+    #[serde(alias = "intermediates")]
+    Intermediates(Vec<PeerId>),
```

Comment:
This should be allowed only in the debug build, not prod.

Reply:
Unfortunately as long as this is needed to make sessions work, it is going to be part of the production build as well